app/src/main/java/com/example/concordia_campus_guide/Models/Building.java

```diff
@@ -21,7 +22,7 @@
 import java.util.List;
 
 @Entity (tableName = "buildings",
-        primaryKeys = {"building_code"}
+        indices = {@Index(value = "building_code", unique = true)}
```

Comment:
why did you change the building_code from being the primary key?

Reply:
If you check out the rest of the code, you will notice that I removed all pk's from entities extending place. Place now uses the id as pk, therefore all those that extend place use the id as pk. this is because we need to call from db with a same method for all (less coupling, and all the same type means it can be stored in the same variable ie: PlaceId)

Reply:
good call 👍